pywwt/annotation.py

```diff
@@ -252,6 +255,100 @@ def _on_trait_change(self, changed):
         super(Line, self)._on_trait_change(changed)
 
 
+class FieldOfView():
+    """
+    A collection of polygon annotations. Takes the name of a pre-loaded 
+    telescope and displays its field of view.
+    """
+    # links for kepler. guide: https://keplerscience.arc.nasa.gov/k2-fields.html
+    # footprint: https://raw.githubusercontent.com/KeplerGO/K2FootprintFiles/master/json/k2-footprint.json
+
+    # more efficient method than CircleCollection of changing trait values?
+    def __init__(self, parent, telescope, center, rot, **kwargs):
+        self.parent = parent
+        self.available = self.parent.instruments.available
+        self.dim = self.parent.instruments.dim
+        
+        # list of IDs of annotations created in this FieldofView instance
+        self.active = []
+        self._gen_fov(telescope, center, rot, **kwargs)
+
+    def _gen_fov(self, telescope, center, rot, **kwargs):
+        #telescope = telescope.lower()
+        if telescope not in self.available:
+            raise ValueError('the given telescope\'s field of view is unavailable at this time')
+
+        position = self.dim[telescope][0]
+        dimensions = self.dim[telescope][-1]
+        # dimensions is a list of lists [ [[ras], [decs]], '', ...]
+
+        # test that pos matches what the user entered
+        if center and position == 'absolute':
+            raise ValueError('the given telescope does not take center coordinates')
+        elif not center and position == 'relative':
+            raise ValueError('the given telescope requires center coordinates')
+
+        # draw jwst nircam's inner panels first
+        if telescope[-6:] == 'nircam':
+            self._gen_fov('nircam_short', center, rot, **kwargs)
+
+        for panel in dimensions:
+            ras = (panel[0] * u.deg).value
+            decs = (panel[1] * u.deg).value
+            ra_r, dec_r = self._rotate(ras, decs, rot)
+
+            if telescope == 'k2':
+                ra_tr = ra_r; dec_tr = dec_r
+            else:
+                center_ra = center.ra.to(u.deg).value
+                center_dec = center.dec.to(u.deg).value
+
+                dec_tr = [center_dec + dec for dec in dec_r]
```

Comment:
Does ``dec_tr = center_dec + dec_r`` not work?

Reply:
`dec_r` is a list of decs for each corner of a panel, so that's why I have to do a list comprehension here. (Also, if I make `dec_r` a length-4 `SkyCoord` and try to add `center_dec`, I get an error saying you can't add a `Quantity` or `int` to a `SkyCoord`.)

I need lists instead of SkyCoords because SkyCoord entries seem to be immutable (`'SkyCoord' object does not support item assignment`) and I need to be able to adjust decs and flip RAs for the `dec` check on line 314.

Reply:
I got confused earlier about where I was taking values. Long story short, I can do array-wise operations now so I don't need the list comprehensions anymore.